src/ansi-c/c_typecheck_type.cpp

```diff
@@ -1469,25 +1469,27 @@ void c_typecheck_baset::typecheck_typeof_type(typet &type)
   c_qualifierst c_qualifiers;
   c_qualifiers.read(type);
 
-  if(!((const exprt &)type).has_operands())
+  const auto &as_expr = (const exprt &)type;
+
+  if(!as_expr.has_operands())
   {
     typet t=static_cast<const typet &>(type.find(ID_type_arg));
     typecheck_type(t);
     type.swap(t);
   }
   else
   {
-    exprt expr=((const exprt &)type).op0();
+    DATA_INVARIANT(
+      as_expr.operands().size() == 1,
+      "typeof of an expression must have one operand");
+    exprt expr = to_unary_expr(as_expr).op();
```

Comment:
Why this `DATA_INVARIANT` here when elsewhere we just use `to_unary_expr`?

Reply:
removed